src/compile.ts

```diff
@@ -269,6 +287,54 @@ export function compile(
         );
       }
 
+      function isNewFunctionlessFunction(
+        node: ts.Node
+      ): node is FunctionInterface {
+        return (
+          ts.isNewExpression(node) &&
+          isFunctionlessClassOfKind(
+            node.expression,
+            Function.FunctionlessType
+          ) &&
+          // only take the form with the arrow function at the end.
+          (node.arguments?.length === 3 || node.arguments?.length === 4) &&
+          ts.isArrowFunction(node.arguments[2])
+        );
+      }
+
+      /**
+       * Heuristically evaluate the fqn of a symbol to be in a module and of a type name.
+       *
+       * /somePath/node_modules/{module}/somePath.{typeName}
```

Comment:
Any other thoughts for handling this heuristically?

Reply:
Structural duck typing. Check for the existence of properties.

Reply:
See https://github.com/functionless/functionless/pull/103#discussion_r884030512

Reply:
What property would you look for on the base CDK construct?

Reply:
I would check for the node property and then check for many of the properties on the node class.

Reply:
Actually, this pattern works for both nested and top level paths (though the comment doesn't mention that).

All it does is check to see if the npm package name and the type name are equal.

Both will return true

`/home/sussmans/functionless/node_modules/constructs/lib/construct.Construct`
and 
`/home/sussmans/functionless/node_modules/aws-cdk-lib/node_modules/constructs/lib/construct.Construct`

I think this will work for moth situations.

Reply:
Cool, sounds good.